src/services/MatchReportService.ts

```diff
@@ -0,0 +1,119 @@
+import ActiveMatchRepository from "../repositories/ActiveMatchRepository";
+import { ActiveMatchTeams, PlayerInActiveMatch } from "../repositories/ActiveMatchRepository/types";
+import LeaderboardRepository from "../repositories/LeaderboardRepository";
+import { UpdatePlayerStatsInput } from "../repositories/LeaderboardRepository/types";
+import { Team } from "../types/common";
+import { waitForAllPromises } from "../utils";
+
+async function calculateProbabilityDecimal(playerInMatchId: string, reportedTeam: Team): Promise<number> {
+  let reportedWinner = 0;
+  let reportedLoser = 0;
+  const teams = await ActiveMatchRepository.getAllPlayersInActiveMatch(playerInMatchId);
+  const blueTeam = teams.blueTeam;
+  const orangeTeam = teams.orangeTeam;
+
+  const blueTeamMMR = blueTeam.reduce((ballchaser1, ballchaser2) => ballchaser1 + ballchaser2.mmr, 0);
+  const orangeTeamMMR = orangeTeam.reduce((ballchaser1, ballchaser2) => ballchaser1 + ballchaser2.mmr, 0);
```

Comment:
So this will work, but I just want to point out something that could cause confusion down the road:
`ballchaser1` isn't actually a BallChaser object. In this scenario, it's actually a number. The first parameter in the callback function for a `.reduce` is the "accumulator." This represents the current value when looping over the array.

I think you already know this, the variable name is just a bit misleading. I think something like `totalMmr` or `accumMmr` might be a bit clearer.

Reply:
Resolved. - Renamed to `totalMMR` and `ballChaser`